tests/blocks/test_block_reference.py

```diff
@@ -0,0 +1,124 @@
+import warnings
+from typing import Dict, Type
+from uuid import uuid4, UUID
+
+import pytest
+from pydantic import ValidationError
+
+from prefect.blocks.core import Block
+from prefect.exceptions import ParameterTypeError
+from prefect.flows import flow
+
+
+class TestBlockReference:
+    class ReferencedBlock(Block):
+        a: int
+        b: str
+
+    class SimilarReferencedBlock(Block):
+        a: int
+        b: str
+
+    class OtherReferencedBlock(Block):
+        c: int
+        d: str
```

Comment:
Could you write a test for `load_from_ref` that includes a nested block document? e.g., 
```python
class NestedBlock(Block):
    child: OtherReferencedBlock
```

Reply:
I am glad you noticed that these are missing. Added few more test for this including nesting with `pydantic.BaseModel` as flow parameter.